### DISCUSSION TITLE
How much of this wishlist (straighten, descreen, crop, etc.) is possible...?

### DISCUSSION THREAD

**Discussion by Karyudo** [ImageMagick/ImageMagick#4780] [Help]
I'm scanning an out-of-print magazine collection, and want to know how much of my 'automatic post-processing' wish list is possible.

In as few steps as possible, I'd like to batch process scanned pages by...
1) straightening/de-skewing the pages automatically (by up to a few degrees)
2) descreening/minimizing moiré of colour and greyscale halftoned images, while leaving text untouched (or at least nicely sharp)
3) saving as 24-bit PNG
4) separating/removing the photos (colour/greyscale) from the pages, saving greyscale text on white background as PNG—preferably with auto-generated image filenames and small marks added to the greyscale text page showing image locations for re-composing
4) storing the removed colour/greyscale photos as JPGs for re-composing

I plan to scan at 200-600 dpi. Finished pages will be 9" wide by 8" high at 192 dpi (1728 x 1536 pixels). All text is black on white background. No photos overlap, although some are full-bleed into either the gutter or edge of the page.

How feasible is this with IM?

Although this is entirely for my own amusement (100% non-commercial, and not only because of copyright issues), I'd strongly consider hiring a competent consultant to save me from adding "learn advanced ImageMagick" to my already-long To Do list for this project. Thanks!

**Comment by snibgo** [comment]
IM has "-deskew" for your wishlist 1. It can save as PNG or JPG.

IM contains low-level tools that can be used to distinguish text from photos, but this isn't trivial. It helps if your magazine pages are well-behaved, eg text is black on white in rectangular blocks, photos are rectangles, and so on.

Similarly descreening; it isn't trivial.

**Comment by jywarren** [comment]
Just noting descreening is maybe... just possible with a few extra steps following guidance at bottom of this page? https://legacy.imagemagick.org/Usage/fourier/#noise_removal

It does involve manually drawing a mask, however.

I would LOVE some simple reusable scripts for halftone removal.

**Reply by fmw42** [reply]
Notch filtering is not easy to do automatically and well. Getting a good mask is the hard part.  I have a notch filter bash unix shell script that is somewhat effective. You might try that.

**Comment by Karyudo** [comment]
Since halftone screens aren't at all random—most of them are well-defined by SWOP (which specifies which colours are at which angle) and have just a small number of common screen frequencies (e.g. 133 lpi, 150 lpi)—would it be possible to create a handful of standard frequency-domain mask presets (e.g. various lines per inch at various dpi), and choose among them with command switches?

More abstractly (and sort of a tangent), I wonder if you could use results of Fourier analysis of screened material to calculate an accurate deskew angle? Not required for what I've asked about, but might be elegant....

**Reply by fmw42** [reply]
One could do the first, but there is often the need to filter more than just for those frequencies.  See my notch filter examples. Also I already have a deskewing algorithm for text images that uses FFT. See my textdeskew script. Both at my web site (http://www.fmwconcepts.com/imagemagick

**Reply by Karyudo** [reply]
> I already have a deskewing algorithm for text images that uses FFT.

Ah! Very cool. I'll have to give that a try. Since I'm scanning magazines, around 75% of the pages have text.

I'll also have to take another look at trying to understand the notch filter examples. Thanks!

**Reply by th1rtyf0ur** [reply]
I realize this thread is nearly 3y old at this point, but for anyone else coming across this in the future, I recommend checking out https://diybookscanner.org/ -- there are loads of setups for both hardware and software that are specialized for bulk scanning of books & other print material, including automatic deskewing, straightening, adjusting levels, etc.
